pkg/reconciliation/reconcile_racks.go

```diff
@@ -2092,16 +2094,152 @@ func (rc *ReconciliationContext) CheckConditionInitializedAndReady() result.Reco
 	return result.Continue()
 }
 
-func (rc *ReconciliationContext) cleanupAfterScaling() error {
-	var err error
+const (
+	podJobIdAnnotation     = "cassandra.datastax.com/job-id"
+	podJobStatusAnnotation = "cassandra.datastax.com/job-status"
+	podJobHandler          = "cassandra.datastax.com/job-runner"
+	jobHandlerMgmtApi      = "management-api"
+
+	podJobCompleted = "COMPLETED"
+	podJobError     = "ERROR"
+	podJobWaiting   = "WAITING"
+)
+
+var (
+	jobRunner chan int = make(chan int, 1)
+)
+
+func (rc *ReconciliationContext) cleanupAfterScaling() result.ReconcileResult {
+	// We sort to ensure we process the pods in the same order
+	sort.Slice(rc.dcPods, func(i, j int) bool {
+		return rc.dcPods[i].UID < rc.dcPods[j].UID
+	})
+
+	for idx, pod := range rc.dcPods {
+		features, err := rc.NodeMgmtClient.FeatureSet(pod)
+		if err != nil {
+			return result.Error(err)
+		}
+
+		if pod.Annotations == nil {
+			pod.Annotations = make(map[string]string)
+		}
 
-	for idx := range rc.dcPods {
-		err = rc.NodeMgmtClient.CallKeyspaceCleanupEndpoint(rc.dcPods[idx], -1, "", nil)
-		if err == nil {
-			break
+		if podJobId, found := pod.Annotations[podJobIdAnnotation]; found {
+			if features.Supports(httphelper.AsyncSSTableTasks) {
+				_, found := pod.Annotations[podJobStatusAnnotation]
+				if !found {
+					// Pod is currently processing something, or has finished processing.. update the status
+					details, err := rc.NodeMgmtClient.JobDetails(pod, podJobId)
+					if err != nil {
+						rc.ReqLogger.Error(err, "Could not get JobDetails for pod", "Pod", pod)
+						return result.Error(err)
+					}
+
+					if details.Id == "" {
+						// This job was not found, pod most likely restarted. Let's retry..
+						delete(pod.Annotations, podJobIdAnnotation)
+						err = rc.Client.Update(context.Background(), pod)
+						if err != nil {
+							return result.Error(err)
+						}
+						return result.RequeueSoon(2)
```

Comment:
Could we go retry the operation at this point without a requeue?

Reply:
Could, but it made the code and logic look a lot less readable - and I don't think it's a very common issue to happen. Especially since if the pod restarted, it probably can't restart the job very soon in any case.

Reply:
I agree about it making the code less readable. I'm good here.